test/encryptedERC20/EncryptedERC20.fixture.ts

```diff
@@ -1,14 +1,39 @@
 import { ethers } from "hardhat";
 
-import type { EncryptedERC20Mintable } from "../../types";
-import { getSigners } from "../signers";
+import type { IEncryptedERC20, TestEncryptedERC20Mintable } from "../../types";
+import { reencryptHandle } from "../reencrypt";
+import { Signers, getSigners } from "../signers";
+import { FhevmInstances } from "../types";
 
-export async function deployEncryptedERC20Fixture(): Promise<EncryptedERC20Mintable> {
+export async function deployEncryptedERC20Fixture(name: string, symbol: string): Promise<TestEncryptedERC20Mintable> {
   const signers = await getSigners();
-
-  const contractFactory = await ethers.getContractFactory("EncryptedERC20Mintable");
-  const contract = await contractFactory.connect(signers.alice).deploy("Naraggara", "NARA", signers.alice.address); // City of Zama's battle
+  const contractFactory = await ethers.getContractFactory("TestEncryptedERC20Mintable");
+  const contract = await contractFactory.connect(signers.alice).deploy(name, symbol);
   await contract.waitForDeployment();
-
   return contract;
 }
+
+export async function reencryptAllowance(
+  signers: Signers,
+  instances: FhevmInstances,
+  user: string,
+  spender: string,
+  token: IEncryptedERC20,
+  tokenAddress: string,
+): Promise<bigint> {
+  const allowanceHandle = await token.allowance(signers[user as keyof Signers], signers[spender as keyof Signers]);
+  const allowance = await reencryptHandle(signers, instances, user, allowanceHandle, tokenAddress);
```

Comment:
yesterday we agreed to replace this "reencryptHandle" with a "reencrypt64" which would be more general, and have one function for each type, also handling unitialized handles and reverting in case of invalid types. Can you address this issue?